src/core/libmaven/mzUtils.cpp

```diff
@@ -7,274 +7,210 @@
-            i = ++j;
-            j = s.find(c, j);
-
-            if (j == string::npos)
-                v.push_back(s.substr(i, s.length( )));
+    std::string makeLowerCase(string &str)
+    {
+        for (size_t i = 0; i != str.length(); ++i) {
+            str[i] = std::tolower(str[i]);
         }
-        if ( v.size() == 0) v.push_back(s);
+        return str;
     }
 
-    void splitNew(const string& s, const string& c, vector<string>& v){
-
-        const char *whole_row = s.c_str();
-        const char *del = c.c_str();
-
-        CsvParser *csvparser = CsvParser_new_from_string(whole_row, del, 0);
-
-        CsvRow *row;
-
-        row = CsvParser_getRow(csvparser);
-        const char **rowFields = CsvParser_getFields(row);
-        for (int i = 0 ; i < CsvParser_getNumFields(row) ; i++) {
-            v.push_back(rowFields[i]);
-        }
+    void split(const string& str, const string& sep, vector<string>& result)
+    {
+        boost::split(result, str, boost::is_any_of(sep));
     }
 
-    void removeSpecialcharFromStartEnd(vector<string>& fields) {
-        for(unsigned int i=0; i < fields.size(); i++ ) {
+    void removeSpecialcharFromStartEnd(vector<string>& fields)
+    {
+        for(size_t i = 0; i < fields.size(); i++) {
             int n = fields[i].length();
-            if (n > 2 && fields[i][0] == '"' && fields[i][n - 1] == '"') {
-                fields[i] = fields[i].substr(1, n - 2);
-            }
-            if (n > 2 && fields[i][0] == '\'' && fields[i][n - 1] == '\'') {
-                fields[i] = fields[i].substr(1, n - 2);
+            if (n > 2 ){
+                if((fields[i][0] == '"' && fields[i][n - 1] == '"') ||
+                    (fields[i][0] == '\'' && fields[i][n - 1] == '\''))
+                        fields[i] = fields[i].substr(1, n - 2);
             }
         }
     }
 
-    char *mystrcasestr(const char *s1, const char *s2) {
-        register const char *s = s1;
-        register const char *p = s2;
-        do {
-            if (!*p) {
-                return (char *) s1;;
-            }
-            if ((*p == *s)
-                    || (tolower(*((unsigned char *)p)) == tolower(*((unsigned char *)s)))
-               ) {
-                ++p;
-                ++s;
-            } else {
-                p = s2;
-                if (!*s) {
-                    return NULL;
-                }
-                s = ++s1;
-            }
-        } while (1);
-    }
-
-    string substituteInQuotedString(const string& s, const string& chars, const string& substitutions ) {
-        string result;
-        for (string::size_type pos = 0; pos < s.size(); ++pos) {
-            char c = s[pos];
-            string::size_type subst_pos = chars.find_first_of(c);
-            if (subst_pos != string::npos) c = substitutions[subst_pos];
-            result += c;
-        }
-        return result;
+    string compareStr(const string str1,const string str2)
```

Comment:
This is a personal preference but if possible avoid using `1`, `2` or other numeric suffixes to distinguish parameters. Give them better names. Especially in this case, when one is supposed to be checked as the substring of the other.